### PR TITLE
Configure CI via github actions

### PR DESCRIPTION
**Pull request by rabbbit**
Mostly based on https://github.com/uber-go/zap/commit/02a233f7b9041b29460a7372f788b324dbd64c4e

Planning to land and iterate? It seem to work fine, at least for the tests.

### CODE REVIEW

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        run: go mod download
+
+      - name: Lint
+        if: matrix.latest
+        run: make lint
+
+      - name: Test
+        run: make test
+
+      # TODO decide whether we want to benchmark on every run
```
Comment (rabbbit) [inline .github/workflows/go.yaml:49]: This is somewhat slow currently - will probably have to skip, and manage manually.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+name: test
+
+on:
+  pull_request:
+    branches: ['*']
+
+jobs:
+  test:
+
```
Comment (cinchurge) [inline .github/workflows/test.yaml:9]: are the blank lines necessary?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    name: Run tests
+
+    runs-on: ubuntu-latest
```
Comment (cinchurge) [inline .github/workflows/test.yaml:12]: is it possible to run this on latest debian as well?

Comment (rabbbit) [reply .github/workflows/test.yaml:12]: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

No. Could run it on a mac, or windows, but I don't think it really matters? It's the same architecture anyway (?)

File: .github/workflows/test.yaml (hunk 2 above)
Comment (rabbbit) [reply .github/workflows/test.yaml:9]: They're not necessary, but it seems to be pattern people follow: https://docs.github.com/en/free-pro-team@latest/actions/quickstart and https://github.com/uber-go/zap/commit/02a233f7b9041b29460a7372f788b324dbd64c4e.

Could remove but it didn't seem horrible?

File: .github/workflows/test.yaml (hunk 3 above)
Comment (cinchurge) [reply .github/workflows/test.yaml:12]: yea it doesn't really matter i guess

**Review by cinchurge — APPROVED** [review]

**Pull request merged**